docs/dokka-presets/scripts/accessibility.js

```diff
@@ -1,3 +1,16 @@
+/**
+ * Check for elements with a navButton class, which indicates the sidebar has finished loading.
+ */
+function dispatchNavigationLoadedEvent() {
+    if (document.querySelectorAll('.navButton').length > 0) {
+        window.dispatchEvent(new Event('navigationLoaded'));
+    } else {
+        setTimeout(dispatchNavigationLoadedEvent, 100);
```

Comment:
question: this is essentially a while loop, does it make more sense as one? also could an infinite loop ever happen here?

Reply:
Sure it could be converted to a while loop. It's possible an infinite loop could happen, that means the sidebar never loaded. I'm not worried about this because it only means the accessibility fixes won't be applied to the (unloaded) sidebar

Reply:
JavaScript delays are asynchronous and non-blocking. A self-sustaining polling mechanism like this is effectively the best way to wait for a condition.

Reply:
@ianbotsf I pushed some code right after your comment, I think this accomplishes the same thing but it's a little simpler to read?

```
async function dispatchNavigationLoadedEvent() {
    while (!document.querySelectorAll('.navButton').length > 0) {
        await new Promise(resolve => setTimeout(resolve, 100));
    }
    window.dispatchEvent(new Event('navigationLoaded'));
}
```

Reply:
Oh yes I suppose that works. 😅 I have no opinions about the relative readability.